### PR TITLE
Improve the docstring of the class SoftDTWLossPyTorch and rename the parameter dist_func into squared_dist_func

### PR DESCRIPTION
**Pull request by YannCabanes**
Improve the docstring of the class `SoftDTWLossPyTorch` and rename the parameter `dist_func` into `squared_dist_func`.

### CODE REVIEW

**Comment by YannCabanes** [comment]
Hello @rtavenar, could you review this PR please?
The failing tests are not related to this PR, I am trying to solve these issues in the PR #468.

File: tslearn/metrics/soft_dtw_loss_pytorch.py [hunk 1]
```diff
@@ -73,32 +73,117 @@ def backward(ctx, grad_output):
             return grad_output.view(-1, 1, 1).expand_as(E) * E, None, None
 
     class SoftDTWLossPyTorch(torch.nn.Module):
-        """Class for the Soft-DTW loss function in PyTorch.
+        r"""Define the Soft-DTW loss function in PyTorch.
```
Comment (rtavenar) [inline tslearn/metrics/soft_dtw_loss_pytorch.py:76]: I would remove the "Define the"

File: tslearn/metrics/soft_dtw_loss_pytorch.py [hunk 2]
```diff
@@ -73,32 +73,117 @@ def backward(ctx, grad_output):
...
             Regularization parameter.
+            It should be strictly positive.
             Lower is less smoothed (closer to true DTW).
         normalize : bool
             If True, the Soft-DTW divergence is used.
             The Soft-DTW divergence is always positive.
             Optional, default: False.
-        dist_func : callable
-            The distance function.
+        squared_dist_func : callable
```
Comment (rtavenar) [inline tslearn/metrics/soft_dtw_loss_pytorch.py:129]: I prefer the previous version for these 2 lines.

**Comment by rtavenar** [comment]
This looks great. I only have two comments and we should be able to merge soon.

File: tslearn/metrics/soft_dtw_loss_pytorch.py (hunk 1 above)
Comment (YannCabanes) [reply tslearn/metrics/soft_dtw_loss_pytorch.py]: Done.

File: tslearn/metrics/soft_dtw_loss_pytorch.py (hunk 2 above)
Comment (YannCabanes) [reply tslearn/metrics/soft_dtw_loss_pytorch.py:129]: I am afraid that the variable name `dist_func` could be confusing for the users.
By default, when `dist_func=None`, `dist_func` is defined as the squared Euclidean distance (`_euclidean_squared_dist`) which is not a distance function, that's why I wanted to rename `dist_func` into `squared_dist_func`.
Moreover we request this callable function to be differentiable whereas distance functions have singular values in which they are not differentiable.
Do you still want to keep `dist_func` for the parameter name @rtavenar ?
If not `squared_dist_func`, could we use another parameter name (like `dissimilarity`)?

Comment (rtavenar) [reply tslearn/metrics/soft_dtw_loss_pytorch.py:129]: I understand your reasons but I still think it should not be called squared... Since one can pick another similarity measure (and having pointwise singularities is not an issue in fact).
Another option would be to call it metric, but since we have released a version in which we called it dist_fun I would rather not break backward compatibility for an issue that seems minor to me. You can maybe edit the docs so that they say "Distance function (or dissimilarity)"

Comment (YannCabanes) [reply tslearn/metrics/soft_dtw_loss_pytorch.py]: Ok, done.

**Comment by YannCabanes** [comment]
All tests using Python 3.10 have failed.

**Comment by YannCabanes** [comment]
All tests using Python 3.10 have failed.
